src/tracks/playabletrack/wavetrack/ui/SpectrumCache.cpp

```diff
@@ -71,16 +71,16 @@ void ComputeSpectrogramGainFactors
 
 }
 
-bool SpecCache::Matches
-   (int dirty_, double pixelsPerSecond,
-    const SpectrogramSettings &settings, double rate) const
+bool SpecCache::Matches(
+   int dirty_, double pixelsPerSecond, const SpectrogramSettings& settings,
+   double sampleRate, double stretchRatio) const
 {
    // Make a tolerant comparison of the pps values in this wise:
    // accumulated difference of times over the number of pixels is less than
    // a sample period.
    const double tstep = 1.0 / pixelsPerSecond;
    const bool ppsMatch =
-      (fabs(tstep - 1.0 / pps) * len < (1.0 / rate));
+      (fabs(tstep - 1.0 / pps) * len < (stretchRatio / sampleRate));
```

Comment:
The commit for pencil tool does some irrelevant things in Spectrum view.

I'd like the minimal changes to maintain compilation to go into the previous commit, and all the more substantial changes attempting to correct spectrum view, in the last one.

Reply:
... But compilation of spectrum code is still broken in this commit anyway.

Reply:
This morning I addressed spectrum drawing in a decent way I think. I squashed those changes and force-pushed. These lines are outdated.